From 639c3a63bcb84242057d4485c96a9ec9513a77c1 Mon Sep 17 00:00:00 2001
From: Tian Le <le.tian@windriver.com>
Date: Tue, 24 Aug 2010 19:33:37 +0800
Subject: [PATCH] octeon: correctly boot either with "mem=xx" arg or not.

support "mem=xx" boot arg. Users can config memory size by using it.
If not use "mem=xx" or configure it as "mem=auto", kernel will
use the full physical memory.

Signed-off-by: Tian Le <le.tian@windriver.com>
---
 arch/mips/cavium-octeon/setup.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index 9a31bff..18db8d2 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -523,6 +523,7 @@ void __init prom_init(void)
 	octeon_boot_desc_ptr = (struct octeon_boot_descriptor *)fw_arg3;
 	octeon_bootinfo =
 		cvmx_phys_to_ptr(octeon_boot_desc_ptr->cvmx_desc_vaddr);
+	MAX_MEMORY = (uint64_t)octeon_bootinfo->dram_size << 20;
 	cvmx_bootmem_init(cvmx_phys_to_ptr(octeon_bootinfo->phy_mem_desc_addr));
 
 	/*
@@ -890,8 +891,7 @@ void __init plat_mem_setup(void)
 	 * regions next to each other.
 	 */
 	cvmx_bootmem_lock();
-	while ((boot_mem_map.nr_map < BOOT_MEM_MAP_MAX)
-		&& (total <= OCTEON_MAX_PHY_MEM_SIZE)) {
+	while ((boot_mem_map.nr_map < BOOT_MEM_MAP_MAX) && (total < MAX_MEMORY)) {
 #if defined(CONFIG_64BIT) || defined(CONFIG_64BIT_PHYS_ADDR)
 		memory = cvmx_bootmem_phy_alloc(mem_alloc_size,
 						__pa_symbol(&__init_end), -1,
-- 
1.6.5.2

